.github/workflows/publish-docs.yml

```diff
@@ -0,0 +1,29 @@
+name: Publish docs via Github pages
+on:
+  push:
+    branches:
+      - master
+      - mkdocs
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout master
+        uses: actions/checkout@v4
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.x
+      - run: echo "cache_id=$(date --utc '+%V')" >> $GITHUB_ENV
+      - uses: actions/cache@v4
+        with:
+          key: mkdocs-material-${{ env.cache_id }}
+          path: .cache
+          restore-keys: |
+            mkdocs-material-
+      - run: pip install \
+          mkdocs-material \
+          pymdown-extensions \
+          mkdocs-glightbox \
+          mkdocs-pymdownx-material-extras
+      - run: mkdocs gh-deploy --force
```

Comment:
Nit:
Does GitHub accept the file extension to be `.yaml` instead of `.yml`? If yes, can the file be renamed to `publish-docs.yaml`?
I ask since all yaml files in the repository use the `.yaml` extension.

Reply:
I have followed the naming convention from all yaml files within .github
See https://github.com/gardener/etcd-druid/blob/master/.github/dependabot.yml

Reply:
Alright. Let's leave this as is for now.